internal/controller/deployment_controller.go

```diff
@@ -319,6 +319,13 @@ func (r *DeploymentReconciler) Delete(ctx context.Context, l logr.Logger, deploy
 		}
 		return ctrl.Result{}, err
 	}
+	if !hr.Spec.Suspend {
+		l.Info("Suspending HelmRelease %s", deployment.Name)
+		hr.Spec.Suspend = true
```

Comment:
If you make it suspended, flux won't remove the helm release and underlying resources